src/Libraries/Microsoft.Extensions.AI.Abstractions/Utilities/AIJsonSchemaTransformOptions.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#pragma warning disable S1067 // Expressions should not be too complex
+
+using System;
+using System.Text.Json.Nodes;
+
+namespace Microsoft.Extensions.AI;
+
+/// <summary>
+/// Provides options for configuring the behavior of <see cref="AIJsonUtilities"/> JSON schema transformation functionality.
+/// </summary>
+public sealed record class AIJsonSchemaTransformOptions
+{
+    /// <summary>
+    /// Gets a callback that is invoked for every schema that is generated within the type graph.
+    /// </summary>
+    public Func<AIJsonSchemaTransformContext, JsonNode, JsonNode>? TransformSchemaNode { get; init; }
```

Comment:
Just to make sure I understand correctly, an end developer could use this TransformSchemaNode to achieve all of the same transformations we handle for the bool props below, right? They're just so common that it's worthwhile for us to provide them with simple knobs?

Reply:
Correct. This is similar to the knobs in `AIJsonSchemaCreateOptions`.